plugins/SpectrumAnalyzer/SaControlsDialog.cpp

```diff
@@ -194,6 +208,109 @@ SaControlsDialog::SaControlsDialog(SaControls *controls, SaProcessor *processor)
 	processor->rebuildWindow();
 	connect(&controls->m_windowModel, &ComboBoxModel::dataChanged, [=] {processor->rebuildWindow();});
 
+	// set stretch factors so that combo boxes expand first
+	config_layout->setColumnStretch(3, 2);
+	config_layout->setColumnStretch(5, 3);
+
+
+	// Advanced configuration
+	QWidget *advanced_widget = new QWidget;
+	QGridLayout *advanced_layout = new QGridLayout;
+	advanced_widget->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Minimum);
+	advanced_widget->setMaximumHeight(m_configHeight);
+	advanced_widget->setLayout(advanced_layout);
+	advanced_widget->hide();
+	controls_layout->addWidget(advanced_widget);
+	controls_layout->setStretchFactor(advanced_widget, 10);
+
+	// Peak envelope resolution
```

Comment:
I assume you are referring to the code that follows this comment?

Not sure there is anything to be gained.. You still need to set label, tool tip, hint text and layout grid position for each knob _somehow_. Some of the code could be wrapped into a function that hides the process of setting individual values, but that would only move all the arguments from separate lines into one huge function call with large argument list where nobody knows what each argument does unless you look up the definition. In my opinion, that would only make it worse.

Another way would be probably to define the layout using XML or whatever Qt uses (never used this kind of layout designer), but that would only move the redundancy to another file, where you still need to define all the attributes anyway.

Reply:
I was indeed thinking about a function (lambda), since I always prefer having less lines in the final code, no matter how hard it will get to read. However, if you think it would get worse here, I'm OK with that, too.

Reply:
The number of lines would probably stay the same, since the argument list would likely go past 80+ characters multiple times, so it would need to be split into multiple lines for better readability anyway.

A lambda or other function would save some time at the beginning (writing name of the current knob 5 times in a row is a pain), but I don't see any value in re-making everything from scratch now. (I actually remember saying the same thing in the original PR, I think you proposed a lambda function back then as well.) And for this PR I didn't want to suddenly start using a different style for half of the code.

Reply:
OK, I'll hit resolve then.